src/imitation/rewards/serialize.py

```diff
@@ -17,33 +23,65 @@
 reward_registry: registry.Registry[RewardFnLoaderFn] = registry.Registry()
 
 
-def _load_reward_net_as_fn(shaped: bool) -> RewardFnLoaderFn:
-    def loader(path: str, venv: VecEnv) -> common.RewardFn:
-        """Load train (shaped) or test (not shaped) reward from path."""
-        del venv  # Unused.
-        net = th.load(str(path))
-        if not shaped and hasattr(net, "base"):
-            # If the "base" attribute exists, we are dealing with a ShapedRewardNet
-            # and will disable the potential shaping (if shaped is False).
-            # If no "base" attribute exists, we seem to use an unshaped RewardNet
-            # anyway, so we just use its predict() method directly.
-            reward = net.base.predict
+def _validate_reward(reward: common.RewardFn) -> common.RewardFn:
+    """Add sanity check to the reward function for dealing with VecEnvs."""
+
+    def rew_fn(
+        obs: np.ndarray,
+        act: np.ndarray,
+        next_obs: np.ndarray,
+        dones: np.ndarray,
+    ) -> np.ndarray:
+        rew = reward(obs, act, next_obs, dones)
+        assert rew.shape == (len(obs),)
+        return rew
+
+    return rew_fn
+
+
+def _strip_wrappers(
+    net: RewardNet,
+    wrappers: List[Type[RewardNetWrapper]],
+) -> RewardNet:
+    """Attempts to remove provided wrappers.
+
+    wrappers until either it reaches a non-warper
+    reward net or it has removed all the listed wrappers.
+
+    Args:
+        net: an instance of a reward network that may be wrapped
+        wrappers: list of wrapper types to remove
+
+    Returns:
+        The reward network with the listed wrappers removed
+    """
+    for wrapper_type in wrappers:
```

Comment:
I would suggest simplying this part using a while loop, similar to [here](https://github.com/HumanCompatibleAI/imitation/blob/f332680115380ce654cfd061fbaaa7bc47918d2c/src/imitation/algorithms/adversarial/airl.py#L98).
```python
# Recursively return the base network of the wrapped reward net
while isinstance(reward_net, reward_nets.RewardNetWrapper):
    reward_net = reward_net.base
```

Though it would be better to add more detailed comments here. WDYT?

Reply:
I guess I was imagining the case where we have a reward network that is shaped and then normalized. Maybe the user just wants to remove the normalization but not the shaping.

Reply:
See https://github.com/HumanCompatibleAI/imitation/issues/437#issuecomment-1136575041

Reply:
That makes sense. Thanks for reminding me of that.